### ISSUE TITLE
failed to read downloaded context: failed to load cache key: invalid response status 403

### ISSUE THREAD

**Issue by atali** [kreuzwerker/terraform-provider-docker#534]
Hello,

I have a weird behaviour, trying to build a simple docker image but it failed with the following error :

```
Terraform will perform the following actions:

  # docker_image.my-image will be created
  + resource "docker_image" "my-image" {
      + id          = (known after apply)
      + image_id    = (known after apply)
      + name        = "my-image:latest"
      + repo_digest = (known after apply)

      + build {
          + cache_from   = []
          + context      = "/EruUkohXdib6p2nB0KVIFDv54yk/HpGfE2XOJY4zO8Sxa1eXbxzHPjk"
          + dockerfile   = "/EruUkohXdib6p2nB0KVIFDv54yk/HpGfE2XOJY4zO8Sxa1eXbxzHPjk/mydocker/Dockerfile"
          + extra_hosts  = []
          + remove       = true
          + security_opt = []
          + tag          = []
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
docker_image.my-image: Creating...
│ 
│ Error: failed to read downloaded context: failed to load cache key: invalid response status 403
│ 
│ 
│ 
│   with docker_image.my-image,
│   on main.tf line 13, in resource "docker_image" "my-image":
│   13: resource "docker_image" "my-image" {
│ 
╵
```

Below the terraform:

```
terraform {
  required_providers {
    docker = {
      source  = "kreuzwerker/docker"
      version = "~> 3.0.2"
    }
  }
}

provider "docker" {
}

resource "docker_image" "my-image" {
  name = "my-image:latest"

  build {

    context = "${path.cwd}"
    dockerfile = "${path.cwd}/mydocker/Dockerfile"
  }
}
```

This is the content of the folder /EruUkohXdib6p2nB0KVIFDv54yk/HpGfE2XOJY4zO8Sxa1eXbxzHPjk:

![image](https://user-images.githubusercontent.com/369628/227510556-3832d0df-f062-4620-a42d-021b38edddbe.png)

when I cd to  /EruUkohXdib6p2nB0KVIFDv54yk/HpGfE2XOJY4zO8Sxa1eXbxzHPjk and build with docker , it works

```
docker build -t my-image:latest ./mydocker
```

My dockerfile :
```
# syntax=docker/dockerfile:1
FROM busybox

RUN echo "SUCCESS"
```
Does someone has an idea ?

**Comment by atali** [comment]
Ok I think I found out the issue ! you need to disable the containerd for pulling and storing images feature in docker desktop

![image](https://user-images.githubusercontent.com/369628/227714227-50ad4298-ed2e-47aa-9e23-a44b6d62740d.png)

**Comment by evilpilaf** [comment]
This saved me after so much searching on why I might be having the problem, any clue if it'll be compatible or why this was?

**Comment by garysassano** [comment]
I just set up Docker Desktop on a new machine, cloned a CDKTF project from my GitHub, and spent two hours troubleshooting the issue.

I've never needed to modify any Docker Desktop settings before, so this must be a feature they've introduced within the past year.

Looking at the [comment above](https://github.com/kreuzwerker/terraform-provider-docker/issues/534#issuecomment-1483798237), it was still listed under `Beta features` as of March 2023. Now, it's moved to `General` and is enabled by default on a fresh installation. Quite frustrating.

![image](https://github.com/user-attachments/assets/ae4b2bb9-a64f-42fa-8820-6c666900f69b)

**Comment by garysassano** [comment]
I recently installed Docker Desktop on a new machine. Today, I cloned a project from my GitHub and tried deploying it, but it wasn't working. Since it had worked before, I was puzzled and ended up spending two hours troubleshooting the issue.

I've never needed to modify any Docker Desktop settings before, so this must be a feature they've introduced within the past year.

Looking at the [comment above](https://github.com/kreuzwerker/terraform-provider-docker/issues/534#issuecomment-1483798237), it was still listed under `Beta features` as of March 2023. Now, it's moved to `General` and is enabled by default on a fresh installation. Quite frustrating.

![image](https://github.com/user-attachments/assets/ae4b2bb9-a64f-42fa-8820-6c666900f69b)
